File: dcp-solutions/2024-10-09/solution.go
```go
func trap(height []int) int {
    n := len(height)
    left, right := 0, n-1
    maxLeft, maxRight := height[0], height[n-1]
    water := 0

    for left < right {
        if maxLeft >= maxRight {
            maxLeft = height[left]
            left++
        } else {
            maxRight = height[right]
            right--
        }
        water += max(maxLeft-maxRight, 0)
    }

    return water
}
```

File: dcp-solutions/2024-10-09/solution.php
```php
<?php
function trap($height) {
    if (empty($height)) return 0;
    
    $left = 0;
    $right = count($height) - 1;
    $max_left = $height[0];
    $max_right = $height[count($height) - 1];
    
    while ($left < $right) {
        if ($max_left <= $max_right) {
            $max_left = max($max_left, $height[$left]);
            $water += max(0, min($max_left, $max_right) - $height[$left]);
            $left++;
        } else {
            $max_right = max($max_right, $height[$right]);
            $water += max(0, min($max_left, $max_right) - $height[$right]);
            $right--;
        }
    }
    
    return $water;
}
?>
```